src/proposals/Proposals.jl

```diff
@@ -248,5 +249,140 @@ function (prop::RStagger)(rng::AbstractRNG,
+        
+        # define starting "window"
+        r = rand(rng) # initial scale/offset
+        u_l = @. u - r * axis # left bound
+        if unitcheck(u_l)
+            v_l = prior_transform(u_l)
+            logl_l = loglike(v_l)
+        else
+            logl_l = -Inf
+        end
+        nc += 1    
+        nexpand += 1
+        
+        u_r = u_l .+ axis # right bound
+        if unitcheck(u_r)
+            v_r = prior_transform(u_r)
+            logl_r = loglike(v_r)
+        else
+            logl_r = -Inf
+        end
+        nc += 1
+        nexpand += 1
+        
+        # stepping out left and right bounds
+        while logl_l >= logl_star
+            u_l .-= axis
+            if unitcheck(u_l)
+                v_l = prior_transform(u_l)
+                logl_l = loglike(v_l)
+            else
+                logl_l = -Inf
+            end
+            nc += 1
+            nexpand += 1
+        end
+        
+        while logl_r >= logl_star
+            u_r += axis
+            if unitcheck(u_r)
+                v_r = prior_transform(u_r)
+                logl_r = loglike(v_r)
+            else
+                logl_r = -Inf
+            end
+            nc += 1
+            nexpand += 1
+        end
+        
+        # sample within limits. If the sample is not valid, shrink the limits until the `logl_star` bound is hit
+        window_init = norm(u_r - u_l) # initial window size
+        while true
+            
+            # define slice and window
+            u_hat = u_r - u_l
+            window = norm(u_hat)
+            
+            # check if the slice has shrunk to be ridiculously small
+            window < 1e-5 * window_init && error("Random slice sampling appears to be stuck.")
+        
+            # propose a new position
+            u_prop = @. u_l + rand(rng) * u_hat  # scale from left
+            if unitcheck(u_prop)
+                v_prop = prior_transform(u_prop)
+                logl_prop = loglike(v_prop)
+            else
+                logl_prop = -Inf
+            end
+            nc += 1
+            ncontract += 1
+            
+            # if success, then move to the new position
+            if logl_prop >= logl_star
+                append!(fscale, window/axlen)
+                u = u_prop
+                break
+            # if fail, then check if the new point is to the left/right of the original point along the proposal axis and update the bounds accordingly
+            else
+                s = dot(u_prop - u, u_hat)    # check sign (+/-)
+                if s < 0    # left
+                    u_l = u_prop
+                elseif s > 0    # right
+                    u_r = u_prop
+                else     # if `s = 0` something has gone wrong
+                    error("Random slice sampler has failed to find a valid point.")
+                end
+            end
+        end # end of sample within limits while       
+        end # end of random slice sampling loop
```

Comment:
all of this code is exactly the same as the other slice sampling method. I would strongly suggest factoring this common functionality out into a separate method that just takes a vector "axis" and axlength and whatever other things and ran this code! saves a lot of space and keeps testing easier.

Reply:
done